src/cargo/ops/cargo_add/mod.rs

```diff
@@ -364,7 +364,24 @@ fn resolve_dependency(
     };
     selected_dep = populate_dependency(selected_dep, arg);
 
-    let old_dep = get_existing_dependency(manifest, selected_dep.toml_key(), section)?;
+    let mut old_dep = get_existing_dependency(manifest, selected_dep.toml_key(), section)?;
+    if old_dep.is_none() && selected_dep.source().is_none() && selected_dep.rename().is_none() {
+        for name_permutation in [
+            selected_dep.name.replace('-', "_"),
+            selected_dep.name.replace('_', "-"),
+        ] {
```

Comment:
Instead of looking up `old_dep` and then looking it up again, could we include the user-selected name in the list of permutations?

Reply:
I will try that, but I guess I have handle the rename case then separately.

Reply:
Looks like you went ahead and did this in 543a138c285cec20f1454af02b4859cd90a03487 without any aspect of rename.  So is that not needed then?

Reply:
I am handling renames in [these lines](https://github.com/rust-lang/cargo/pull/13765/files#diff-7d08d5b8dd5fdeaca6bb19b814d06790d42f63f052b42a41f3eb5cf8ec89456cR463-R465) by not doing fuzzy lookups when the dependency is renamed